### PR TITLE
[0.3.4]: Test/termination abstract_round abci

### PR DESCRIPTION
**Pull request by 0xArdi**
## Proposed changes

This PR adds the termination related tests to `abstract_round_abci`.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [x] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [x] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: packages/valory/skills/abstract_round_abci/test_tools/abci_app.py [hunk 1]
```diff
@@ -79,6 +80,77 @@ def process_payload(self, payload: BaseTxPayload) -> None:
...
+
+    def end_block(self) -> None:
+        """End block."""
+
+    def check_payload(self, payload: BaseTxPayload) -> None:
+        """Check payloads of type 'payload_background_c'."""
+
+    def process_payload(self, payload: BaseTxPayload) -> None:
+        """Process payloads of type 'payload_background_c'."""
+
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/test_tools/abci_app.py:97]: could be moved to base class

File: packages/valory/skills/abstract_round_abci/tests/test_base.py [hunk 2]
```diff
@@ -1790,6 +1912,99 @@ def test_update_round(
...
+                == self.round_sequence.root_hash
+            )
+            assert (
+                self.round_sequence._last_round_transition_tm_height
+                == self.round_sequence.tm_height
+            )
+            process_event_mock.assert_called_with(
+                background_round_result[-1],
+                result=background_round_result[0],
+            )
```
Comment (Karrenbelt) [inline packages/valory/skills/abstract_round_abci/tests/test_base.py:2006]: what I dislike about the current implementation is that there is no `else` clause. 

```
same_timestamp = self.round_sequence._last_round_transition_timestamp
                != self.round_sequence._blockchain.last_block.timestamp
same_height =  self.round_sequence._last_round_transition_height
                != self.round_sequence._blockchain.height
same_root_hash = self.round_sequence._last_round_transition_root_hash
                != self.round_sequence.root_hash
same_tm_height = self.round_sequence._last_round_transition_tm_height
                != self.round_sequence.tm_height
                
conditions_are_equal = (same_timestamp, same_height, same_root_hash, same_tm_height)
                
if background_round_result is current_round_result is None:
    assert not any(conditions_are_equal)
    process_event_mock.assert_not_called()
else:  # ?
     assert all(conditions_are_equal)
     process_event_mock.assert_called_with(
         current_round_result[-1],
         result=current_round_result[0],
     )
```

Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/tests/test_base.py:2006]: The case when `background_round_result is None` is covered already in `test_update_round`.

File: packages/valory/skills/abstract_round_abci/test_tools/abci_app.py (hunk 1 above)
Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/test_tools/abci_app.py:97]: As in define a base concrete class?

File: packages/valory/skills/abstract_round_abci/tests/test_base.py (hunk 2 above)
Comment (0xArdi) [reply packages/valory/skills/abstract_round_abci/tests/test_base.py:2006]: But I understand what you mean. We could refactor this and have it all in one method.

Comment (Karrenbelt) [reply packages/valory/skills/abstract_round_abci/tests/test_base.py:2006]: even if it cannot happen, then I'd rather have an `else: raise` so another dev knows that something won't ever silently pass

File: packages/valory/skills/abstract_round_abci/test_tools/abci_app.py (hunk 1 above)
Comment (Karrenbelt) [reply packages/valory/skills/abstract_round_abci/test_tools/abci_app.py:97]: just to keep it dry. 
```
BaseTerminationRound(AbstractRound):
   ...
```

**Review by DavidMinarsch — APPROVED** [review]

**Pull request merged**